### ISSUE TITLE
twisted.python.sendmsg segfault on Linux + Python 2

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8969]
|[<img alt="exvito's avatar" src="https://avatars.githubusercontent.com/u/7661068?s=50" width="50" height="50">](https://github.com/exvito)| @exvito reported|
|-|-|
|Trac ID|trac#8969|
|Type|defect|
|Created|2016-12-29 16:28:59Z|

While working on [#8912](https://github.com/twisted/twisted/issues/8912) and the associated github PR I hit a segmentation fault on a test I was preparing.

A minimal code sample that triggers it is attached.

I think narrowed it down Linux + Python 2. On my tests, it works fine on Linux + Python 3, and fails with socket.error/OSError on Mac OS 10.9.5 + Python 2/3. No BSDs at hand to try out...

As far as I can tell, no code in Twisted uses sendmsg in the way the sample is using it. However, twisted.python.sendmsg is a public API so someone somewhere may hit this like I just did.

Attachments:

* [tx_py27_sendmsg_segfault.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/4c0/4c00ef5491a867fdbd9000315ffc9e13a3c20049/027087e73f26c5dd64a02fd118f92b2663a12781.py) (943 bytes) - added by exvito on 2016-12-29 16:30:18Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__8969 8969
type__defect defect
reporter__exvito exvito
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1483028939106188 1483028939106188
changetime__1500827708228314 1500827708228314
version__None None
owner__Craig_Rodrigues__rodrigc_____ Craig Rodrigues <rodrigc@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exvito's avatar" src="https://avatars.githubusercontent.com/u/7661068?s=50" width="50" height="50">](https://github.com/exvito)<a name="note_1"></a>|@exvito commented|
|-|-|

Github PR at https://github.com/twisted/twisted/pull/650.

**Comment by twisted-trac** [comment]
|[<img alt="exvito's avatar" src="https://avatars.githubusercontent.com/u/7661068?s=50" width="50" height="50">](https://github.com/exvito)<a name="note_2"></a>|@exvito commented|
|-|-|

As noted by rodrigc in the PR, this may fix [#5649](https://github.com/twisted/twisted/issues/5649).

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @rodrigc<br>@rodrigc set status to `closed`|
|-|-|

In changeset d1b91b92516aa23b52de68d592c1edd30f757072
```
#!CommitTicketReference repository="" revision="d1b91b92516aa23b52de68d592c1edd30f757072"
Merge pull request #651 from twisted/8969-exvito-sendmsg-segfault

Author: exvito
Reviewer: rodrigc
Fixes: ticket:8969

Fix for twisted.internet.sendmsg segfault on Linux Python 2
```
